src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -1844,3 +1870,36 @@ class FileReferenceSegment(BaseSegment):
         # to match as a `TableReferenceSegment`
         Ref("QuotedIdentifierSegment"),
     )
+
+
+@spark3_dialect.segment(replace=True)
+class FromExpressionElementSegment(BaseSegment):
+    """A table expression.
+
+    Enhanced from Ansi to allow for `LATERAL VIEW` clause
+    """
+
+    type = "from_expression_element"
+    match_grammar = Sequence(
+        Ref("PreTableFunctionKeywordsGrammar", optional=True),
+        OptionallyBracketed(Ref("TableExpressionSegment")),
+        AnyNumberOf(
+            Ref("LateralViewClauseSegment"),
+            min_times=1,
+            optional=True,
+        ),
```

Comment:
Are these not contradictory? If it's optional isn't `min_times` actually `0`?

And then the optionality is implicit in `AnyNumberOf` I think.

```suggestion
        AnyNumberOf(Ref("LateralViewClauseSegment")),
```

Reply:
Did you miss this one @R7L208 ?